internal/provider/resource_vault_cluster.go

```diff
@@ -0,0 +1,153 @@
+package provider
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// defaultClusterTimeout is the amount of time that can elapse
+// before a cluster read operation should timeout.
+var defaultVaultClusterTimeout = time.Minute * 5
+
+// createUpdateTimeout is the amount of time that can elapse
+// before a cluster create or update operation should timeout.
+var createUpdateVaultClusterTimeout = time.Minute * 35
+
+// deleteTimeout is the amount of time that can elapse
+// before a cluster delete operation should timeout.
+var deleteVaultClusterTimeout = time.Minute * 25
+
+func resourceVaultCluster() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The Vault cluster resource allows you to manage an HCP Vault cluster.",
+		CreateContext: resourceVaultClusterCreate,
+		ReadContext:   resourceVaultClusterRead,
+		UpdateContext: resourceVaultClusterUpdate,
+		DeleteContext: resourceVaultClusterDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create:  &createUpdateVaultClusterTimeout,
+			Update:  &createUpdateVaultClusterTimeout,
+			Delete:  &deleteVaultClusterTimeout,
+			Default: &defaultVaultClusterTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"cluster_id": {
+				Description:      "The ID of the HCP Vault cluster.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"hvn_id": {
+				Description:      "The ID of the HVN this HCP Vault cluster is associated to.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// TODO: this is required for consul, but is this complete for vault yet? we can also mark computed output until it's ready.
+			"tier": {
+				Description: "The tier that the HCP Vault cluster will be provisioned as.  Only 'development' and 'standard' are available at this time.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				// TODO: generalize consul enum validator
+				// ValidateDiagFunc: validateStringInSlice(vaultClusterResourceTiers, true),
+				DiffSuppressFunc: func(_, old, new string, _ *schema.ResourceData) bool {
+					return strings.ToLower(old) == strings.ToLower(new)
+				},
+			},
+			// optional fields
+			"public_endpoint": {
+				Description: "Denotes that the cluster has a public endpoint for the Vault UI. Defaults to false.",
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"namespace": {
+				Description: "The name of the customer namespace.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				// TODO Confirm:
```

Comment:
Does a new namespace mean new cluster?

Reply:
Yes it would. We don't have a way to rename the namespace for the customer right now. I don't believe there is a way to rename a namespace in Vault.

Reply:
Is there a default name it gets set to? I could include that as a computed output.

Reply:
TODO: investigate cloud-ui's default namespace

Reply:
@thorfour  if we allow Customers to set the name in TF, will that break any backend processes that are assuming the default namespace we assign ?

Reply:
Confirmed with Thor today that `namespace` can't be changed by the user, so I bumped that down to computed outputs.